sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1Test.java

```diff
@@ -267,17 +338,27 @@ public void testWriteDisplayData() {
   public void testDeleteEntityDoesNotAllowNullProject() throws Exception {
     thrown.expect(NullPointerException.class);
     thrown.expectMessage("projectId");
+    DatastoreIO.v1().deleteEntity().withProjectId((String) null);
+  }
 
-    DatastoreIO.v1().deleteEntity().withProjectId(null);
+  @Test
+  public void testDeleteEntityDoesNotAllowNullProjectValueProvider() throws Exception {
+    thrown.expect(NullPointerException.class);
+    thrown.expectMessage("projectId ValueProvider");
+    DatastoreIO.v1().deleteEntity().withProjectId((ValueProvider<String>) null);
   }
 
   @Test
   public void testDeleteEntityValidationFailsWithNoProject() throws Exception {
     DeleteEntity deleteEntity = DatastoreIO.v1().deleteEntity();
+    thrown.expect(NullPointerException.class);
+    thrown.expectMessage("projectId ValueProvider");
+    deleteEntity.validate(null);
 
+    deleteEntity = DatastoreIO.v1().deleteEntity()
```

Comment:
This code will also not run. Please audit the rest of the PR for similar errors.